### ISSUE TITLE
Wrap up 1.84 release (previously 1.83.1)

### ISSUE THREAD

**Issue by w7sst** [w7sst/MorseRunner#233]
Our next release, ~~v1.83.1~~ v1.84, adds the IARU HF Contest and lots of bug fixes to address issues found in the past year.

This task is to track the remaining work, including testing and release notes. Please add additional Tasks below if I'm missing something.

## Update March 16, 2024...
- Here is the final [v1.84-rc4 build](https://1drv.ms/u/s!AiN4lELeOz01q34X8LA963GoLo2k?e=Ugj3LE). I have no other planned changes to make a this time. Please give it a spin as you have time and report back with any test results.
- The last change was to improve the call history distribution for ARRL FD. See #285 and #286 for details.
- Let's target March 20 to let this out the door.

Thank you. 73, Mike

@scotthibbs @K6OK @f6fvy @jr8ppg @WR7Q @swalberg

## Update March 8, 2024...
- Here is [v1.84-rc3](https://1drv.ms/u/s!AiN4lELeOz01q26kvcYUBJodbPBA?e=ak1b3e). Please give the following changes a try:
- Start/Stop button, #281 
- Editable Exchange for CQ WPX, #278 
- HST error checks, #275 

I will be away from my laptop for a few days. I'll try to respond via my phone if i can. Hopefully we can get this release by 3/15 or so. Thank you for everyone's help.

@scotthibbs, we'll need to work on the release notes to get some of these latest changes added to it. I still have to update the Release page. I'll get to that next.

@scotthibbs @K6OK @f6fvy @WR7Q @jr8ppg

## Update March 2, 2024...
- Here is [v1.84-rc2](https://1drv.ms/u/s!AiN4lELeOz01q2Rc28oY_VMSPpGQ?e=JXVzRa). Please give the Serial NR change a spin using the CQ WPX contest.
- It took longer than I thought to code up the Serial NR changes, but I'm very pleased with how it turned out.
- Scott has wrapped up release notes.
- I have to update the Release page on last time.
- We should be getting close to wrapping this one up.

## Update Feb 21, 2024...
- [Here](https://1drv.ms/u/s!AiN4lELeOz01q1bYfmH8YVW2-9KH?e=XWyb7J) is the latest release-candidate build.
- I think we are getting close to a code freeze on this release. I think I've fixed all that I plan to fix.
- Scott is wrapping up the Release Notes, `Readme.txt` file.

## Tasks
- [x] Build pre-release to give to testers
- [x] Write initial release notes (will be given to Scott)
- [x] update copyright year to 2024
- [x] Write initial changes to `Readme.txt` file (part of the installation directory). Initial changes are checked into mainline.
- [x] (Mike) Create release branch and add label
- [x] (Scott) Finalize release notes
- [x] (Mike) Finalize [v1.84 Release page](https://github.com/w7sst/MorseRunner/releases)
- [x] (Scott) Finalize [v1.84 Release page](https://github.com/w7sst/MorseRunner/releases)
- [x] testing by volunteer testers. We'll ask testers to provide feedback directly in github or via email to Scott and Mike. I'm hoping Scott can track this testing detail.
- [x] (Mike) Re-open all issues and add a `fixed ready for validation` label to each of them. We'll assign these out to specific people for testing. GitHub testers will be assigned to this task and to the individual Issue being tested.
- [x] Testers will update each individual issue with comments. Once testing is complete, tester will update label to `verified`.
- [x] Verify new IARU HF Contest
- [x] Verify mouse wheel behavior for RIT and Cntl-key/mouse wheel adjustment of Bandwidth.
- [x] Verify any of the [remaining open issues](https://github.com/w7sst/MorseRunner/milestone/4) for this milestone. Please update the issue with a test report/comment and notify Scott or Mike.
- [x] Scott/Mike will review test report for each `verified` issues and close the issue.
- [x] (Scott) monitor testing progress and update release notes are complete for that item.
- [x] (Mike) fix any issues found during testing and provide additional pre-release builds as needed.
- [x] Write up an announcement for v1.84 that will be announced on GitHub and groups.io.
- [x] What am I missing???
- [x] Merge remaining pull requests.
- [x] (Mike) Provide final build and release image. Publish the release and relax.

**Comment by w7sst** [comment]
Hi @scotthibbs 
- My initial change to `Readme.txt` is shown below.
- You can edit this file [here](https://github.com/w7sst/MorseRunner/blob/main/Readme.txt) or edit it locally on your computer with a `git pull origin` command.
- You will see similar comments in the v1.83.1 Release Draft [here](https://github.com/w7sst/MorseRunner/releases). Click the pencil icon in the upper-right corner to edit the Release Draft. After editing, click the `Save Draft` button.

The initial changes are:
```
Version 1.83.1 (February 2024)
  - bug fix release
  - update call history files for ARRL DX, ARRL FD, CQ WW, CWOPS CWT, K1USN SST and NCJ NAQP Contests
  - improve pattern matching for DXCC entities (used in status bar) (Coded by W7SST)
  - CQ WPX - Dx Stations will occasionally send a serial number of zero (Coded by W7SST)
  - ARRL DX - incorrect handling of KH6/KL7 stations using AH6/AL7, NH6/NL7, WH6/WL7 (Coded by W7SST)
  - NAQP - Exchange field does not allow numbers (e.g. KH6 or KL7) (Coded by W7SST)
  - All Contests - WPM keyboard entry incorrect behavior for Spin Box (up down control) (Coded by W7SST)
  - CWOPS CWT - Contest Parser Reading First Line of File (Coded by W7SST)
  - All Contests - spacebar or Tab will now select both exchange fields (Coded by W7SST)
  - All Contests - Hide Dx Station's Entity status string if same as user's Entity (Coded by W7SST)
  - K1USN SST - user test field in call history file should be optional (Coded by W7SST)
  - DX station will send an abbreviated exchange number in the JARL ALLJA and ACAG contests (Coded by JR8PPG)
  - User's exchange number is not abbreviated (not convert 100 to 1TT) (Coded by JR8PPG)
```

Just a thought, but this may be too much detail for this `Readme.txt` file. Not sure. Perhaps the detail can go in the Release page and the detailed `What's Changed` section of the Release page.

**Comment by w7sst** [comment]
Hi @scotthibbs ,
I created the release branch and updated the Release page with the auto-generated list of changes at the end of the page. It also looks like some of the changes are getting validated by others. Take care, Mike

**Comment by scotthibbs** [comment]
Hi Mike,

I need a day or so. I got my laptop in parts to replace the keyboard but
will try to get it done in the next few days.

**Comment by scotthibbs** [comment]
Mike,

Got my laptop done. Does the draft release have the new contest? And one
question: Does this "User's exchange number is not abbreviated (not convert
100 to 1TT) (Coded by JR8PPG)" mean the exchange will or will not be
abbreviated now?

73
KD4SIR
Scott Hibbs
812-581-0631

**Comment by w7sst** [comment]
Hi Scott,
See #206 for details from jr8ppg's bug fix. The change is to send the user's exchange and not apply random abbreviations to it. On the Dx stations will have random abbreviations added to what they send. Please reach out to jr8ppg if you have any questions.

No, the new contest is not in the pre-release yet.

I'll try to get a pre-release out for 1.84 with the new contest.

-Mike

**Comment by w7sst** [comment]
We will include the IARU HF World Championship contest along with the bug fixes and call it 1.84. Hopefully we can wrap this up before the end of the month. I think our challenge will be to find help validating the changes and testing the new contest.

A copy of the 1.84 pre-release build is available [here](https://1drv.ms/u/s!AiN4lELeOz01q0LYuPt75uEGNMEM?e=rryom5).

**Comment by w7sst** [comment]
Hi all,
[Here is a copy of the latest pre-release v1.84-pr4 build](https://1drv.ms/u/s!AiN4lELeOz01q1F77aAb4wDi3h_z?e=3Oq49i). This includes Issues #248 and #252 currently in code review under #253.

Please provide any testing feedback if you are able to do some testing. There are several issues marked as `Fixed - ready for validation` that have not yet been verified. You can see the list of remaining issues to be closed [here](https://github.com/w7sst/MorseRunner/milestone/4). If you are able to test/verify any of these, please let Scott or myself know. Thank you.

@scotthibbs

**Comment by scotthibbs** [comment]
I tested the 1.84-pr4 tonight and worked the HST for an hour. I used the Run dropdown and selected HST without changing anything. I got a score pop up at the end of the contest but nothing in the File/View Score Table. It states "Your score table is empty". I have a "HstResults.txt" file in the directory and it has two previous entries where I didn't finish (the first is "2024-02-19 21:19 VE3NEA 0"). If I finish this contest early, I get another entry like the first but my score table is still empty. 

I finished an IARU HF competition for 10 minutes and got a verified score of 130 (world record! hihi). All seems well. 

Let me know if I need to create an issue for the high score missing.

**Comment by w7sst** [comment]
> Let me know if I need to create an issue for the high score missing.

Yes, we should create an Issue for this. Thank you.

**Comment by w7sst** [comment]
Hi Scott @scotthibbs ,
It is looking like our request for some extra testing help to verify specific Issues did not pan out. I suggest we reclose those issues where I had reopened the Issue and asked the original submitter to assist in verification. Any thoughts on this? Tnx, Mike

**Comment by w7sst** [comment]
@scotthibbs @K6OK @f6fvy @jr8ppg @WR7Q 
I'm not sure if anyone sees these comments or not. I'd like to wrap up this release by the end of the month. I think we are getting close.

- [Here](https://1drv.ms/u/s!AiN4lELeOz01q1bYfmH8YVW2-9KH?e=XWyb7J) is the latest release-candidate build (v1.84-rc1).
- I think we are getting close to a code freeze on this release. I think I've fixed all that I plan to fix.
- Scott is wrapping up the Release Notes, `Readme.txt` file.
- If anyone has time for some additional testing, we would appreciate it. Please add a comment to this task showing what you have tested. I'll add a list of test items to the task list above. Please mark them off if you have tested/verified an item.
- Also look at the [open issues list](https://github.com/w7sst/MorseRunner/milestone/4) for any items that need to be verified. These are marked as `fixed - ready for validation`. Let us know if you verified an Issue by adding a comment to the Issue.
 
Thank you, Mike

**Comment by scotthibbs** [comment]
I didn't see any more fixed waiting to be verified issues. 
I think we are a go for release. 
Let me know if I missed something.

**Comment by w7sst** [comment]
Thanks Scott,
You can close #218 and #219 as they are both marked as verified. I'm still coding on the CQ Wpx Serial Number change. Should wrap it up this weekend. I have one merge yet to do on mainline. Thanks for the help, Mike

**Comment by w7sst** [comment]
## Update March 8, 2024...
- Here is [v1.84-rc3](https://1drv.ms/u/s!AiN4lELeOz01q26kvcYUBJodbPBA?e=ak1b3e). Please give the following changes a try:
- Start/Stop button, #281 
- Editable Exchange for CQ WPX, #278 
- HST error checks, #275 

I will be away from my laptop for a few days. I'll try to respond via my phone if i can. Hopefully we can get this release by 3/15 or so. Thank you for everyone's help.

@scotthibbs, we'll need to work on the release notes to get some of these latest changes added to it. I still have to update the Release page. I'll get to that next.

@scotthibbs @K6OK @f6fvy @WR7Q @jr8ppg

**Comment by f6fvy** [comment]
Editable Exchange for CQ WPX works fine for me.

As Scott mentioned in #281, the Run/Stop button could be improved. The current behavior (rc3) is nevertheless very usable and is already an improvement over the current version (and older versions). Whether or not this button will be completely redesigned for 1.84 depends on when it is scheduled for official release. The WPX CW contest takes place at the end of May, and I think that many users will greatly appreciate the improvements on this contest for training.

**Comment by WR7Q** [comment]
I love it...work great.

73,
Bob WR7Q
CWA Admin

**Comment by w7sst** [comment]
Hi Scott,
- I'm done with Readme.txt changes (they have been sent to you via a pull request and separate Issue #288 and pull request #289).
- I finished my edits to the [1.84 Release page](https://github.com/w7sst/MorseRunner/releases). Notice that I did change the title to include CQ WPX. What do you think, does this make sense given that we have made several improvements to this contest?
- Let me know how your edits are going.
- You'll also notice a code review for the FD home/portable club station reduction. The code comment describes what we did and why.

Otherwise, I'm ready to go for a release. I see no other changes necessary. Bob (@WR7Q) seems to be satisfied as are Larry (@f6fvy) and Jim (@K6OK).

Oh, we also need to write up an announcement page. I'll work on this tomorrow and send you a draft. If anyone has any creative ideas for an announcement in groups.io, please write it up and send it to me. Thanks.

73, Mike

**Comment by w7sst** [comment]
Concerning the Announcement in groups.io, this is what was posted last year for 1.83...

> ### Version 1.83 (March 19, 2023)
> - Added JARL All Japan Contest (Coded by JR8PPG)
> - Added JARL All Cities All Guns (ACAG) Contest (Coded by JR8PPG)
> - Added K1USN Slow Speed Test (Coded by W7SST)
> - Add support for CWOPS CWT nonmember exchanges (Coded by W7SST)
> - Improve CW spacing and timing (Found by John K3TN, Coded by W7SST)
> 
> To download this release, visit https://github.com/w7sst/MorseRunner/releases/tag/v1.83, scroll down to the Assets section, and download the [Morse.Runner.1.83.zip](https://github.com/w7sst/MorseRunner/releases/download/v1.83/Morse.Runner.1.83.zip) file.

I think for 1.84, we can do something similar (taken from our [Release page](https://github.com/w7sst/MorseRunner/releases)).
### Version 1.84 (March 2024)
  - Added IARU HF World Championship Contest (W7SST)
  - CQ WPX improvements, including editable exchange and improved serial NR generation (W7SST)
  - Update call history files for ARRL DX, ARRL FD, CQ WW, CWOPS CWT, K1USN SST and NCJ NAQP Contests

#### Contest-specific improvements...
  - ARRL DX - incorrect handling of KH6/KL7 stations using AH6/AL7, NH6/NL7, WH6/WL7 (Coded by W7SST)
  - ARRL FD - reduce high number of home/portable club stations (W7SST)
  - CQ WPX - Editable exchange field with starting number or '#' (W7SST)
  - CQ WPX - Rename NR Digits to Serial NR; improve serial number generation (W7SST)
  - CQ WPX - Dx Stations will occasionally send a serial number of zero (Coded by W7SST)
  - CQ WPX - Improved Serial Number generation; NR Digits menu renamed to Serial NR (Coded by W7SST)
  - CWOPS CWT - Contest Parser Reading First Line of File (Coded by W7SST)
  - JARL - DX station will send an abbreviated exchange number in the JARL ALLJA and ACAG contests (Coded by JR8PPG)
  - JARL - User's exchange number is not abbreviated (not convert 100 to 1TT) (Coded by JR8PPG)
  - K1USN SST - user test field in call history file should be optional (Coded by W7SST)
  - NAQP - Exchange field does not allow numbers (e.g. KH6 or KL7) (Coded by W7SST)

#### General bug fixes and improvements...
  - After a CQ, if nothing is caught or if what you caught is wrong, you can use F7 ('?') to instruct the callers to call again. (F6FVY)
  - New callers can be added when a QSO is done. (F6FVY)
  - When F1 was hit several times in a row, the number of callers was increasing without respect to the activity setting. (F6FVY)
  - The "reply or not" algorithm has been modified to allow callers with long call signs to reply, even if only few letters are caught. (F6FVY)
  - Improve Run button with default Run Mode and Stop icon (W7SST)
  - No RIT needed for Single Call Modes (W7SST)
  - Receive Bandwidth adjustment using Cntl-key and mouse wheel (F6FVY, W7SST)
  - Speed increments / decrements modified (2 WPM instead of 5). (F6FVY, W7SST)
  - Improve RIT adjustment using mouse wheel (W7SST)
  - Improve pattern matching for DXCC entities (used in status bar) (Coded by W7SST)
  - WPM keyboard entry incorrect behavior for Spin Box (up down control) (Coded by W7SST)
  - spacebar or Tab will now select both exchange fields (Coded by W7SST)
  - Hide Dx Station's Entity status string if same as user's Entity (Coded by W7SST)
  - Shift-F9 added as shortcut to start Single Call mode (Coded by W7SST)
  - WPM keyboard entry incorrect behavior for Spin Box (up down control) (Coded by W7SST)

This might be too wordy for a general announcement to our respective clubs and news letters and twitter feeds. Any ideas are welcome.  @f6fvy @K6OK @WR7Q @scotthibbs

**Comment by w7sst** [comment]
Here is my proposal for an Announcements page. All I did was add a simple introduction paragraph...

Announcement - Morse Runner Community Edition - Version 1.84.

We are excited to announce the latest release of Morse Runner Community Edition. It adds the upcoming IARU HF World Championship Contest, improvements to the CQ WPX Contest and numerous functional and quality improvements.

Version 1.84 (March 2024)
  - Added IARU HF World Championship Contest (W7SST)
  - CQ WPX improvements, including editable exchange and improved serial NR generation (W7SST)
  - Update call history files for ARRL DX, ARRL FD, CQ WW, CWOPS CWT, K1USN SST and NCJ NAQP Contests

  Contest-specific improvements...
  - ARRL DX - Incorrect handling of KH6/KL7 stations using AH6/AL7, NH6/NL7, WH6/WL7 (Coded by W7SST)
  - ARRL FD - Reduce high number of home/portable club stations (W7SST)
  - CQ WPX - Editable exchange field with starting number or '#' (W7SST)
  - CQ WPX - Improved Serial Number generation; NR Digits menu renamed to Serial NR (Coded by W7SST)
  - CQ WPX - Dx stations will occasionally send a serial number of zero (Coded by W7SST)
  - CWOPS CWT - Contest Parser Reading First Line of File (Coded by W7SST)
  - JARL - DX station will send an abbreviated exchange number in the JARL ALLJA and ACAG contests (Coded by JR8PPG)
  - JARL - User's exchange number is not abbreviated (not convert 100 to 1TT) (Coded by JR8PPG)
  - K1USN SST - User test field in call history file should be optional (Coded by W7SST)
  - NAQP - Exchange field does not allow numbers (e.g. KH6 or KL7) (Coded by W7SST)

  General bug fixes and improvements...
  - Improve Run button with default Run Mode and Stop icon (W7SST)
  - No RIT needed for Single Call Modes (W7SST)
  - After a CQ, if nothing is caught or if what you caught is wrong, you can use
    F7 ('?') to instruct the callers to call again. (F6FVY)
  - New callers can be added when a QSO is done. (F6FVY)
  - When F1 was hit several times in a row, the number of callers was
    increasing without respect to the activity setting. (F6FVY)
  - The "reply or not" algorithm has been modified to allow callers with long
    callsigns to reply, even if only few letters are caught. (F6FVY)
  - Receive Bandwidth adjustment using Cntl-key and mouse wheel (F6FVY, W7SST)
  - Speed increments / decrements modified (2 WPM instead of 5). (F6FVY, W7SST)
  - Improve RIT adjustment using mouse wheel (W7SST)
  - Improve pattern matching for DXCC entities (used in status bar) (Coded by W7SST)
  - WPM keyboard entry incorrect behavior for Spin Box (up down control) (Coded by W7SST)
  - Spacebar or Tab will now select both exchange fields (Coded by W7SST)
  - Hide Dx Station's Entity status string if same as user's Entity (Coded by W7SST)
  - Shift-F9 added as shortcut to start Single Call mode (Coded by W7SST)

@scotthibbs @K6OK @WR7Q @f6fvy @jr8ppg

**Comment by f6fvy** [comment]
Excellent. Thanks

**Comment by scotthibbs** [comment]
I'm happy with everything and report go for launch.

**Comment by richqck** [comment]
I have been testdriving the various RC's and the program hasn't randomly hung like it did with 1.83.  Whatever you did, must have been the fix. Thanks!

**Comment by w7sst** [comment]
## Update March 16, 2024...
- Here is the final [v1.84-rc4 build](https://1drv.ms/u/s!AiN4lELeOz01q34X8LA963GoLo2k?e=Ugj3LE). I have no other planned changes to make a this time. Please give it a spin as you have time and report back with any test results.
- The last change was to improve the call history distribution for ARRL FD. See #286 for details.
- Let's target March 20 to let this out the door.

Thank you. 73, Mike

@scotthibbs @K6OK @f6fvy @jr8ppg @WR7Q @swalberg

**Comment by scotthibbs** [comment]
I like the announcements too! I'm ready and still go for launch.

**Comment by K6OK** [comment]
I tested 1.84 rc4 on both WPX and FD.  Works very well, no issues found.   Announcement looks good. I concur with "ready to release."

One comment about FD: I think the pendulum was swung too far and 1D's are too rare.  It doesn't match what is heard on the air.  However, that said, I like the extra emphasis on 1A, 2A, 1E, 2A etc. because that's what FD is really about.  I think most users will prefer it this way.  I support the FD call history distribution as you have it in rc4, even if it strays from reality a bit.  Contest simulation with artistic license ;-)

73 Jim K6OK

@scotthibbs @WR7Q @f6fvy @jr8ppg

**Comment by f6fvy** [comment]
All OK for me. Tnx all for your work.

**Comment by w7sst** [comment]
Thanks everyone for helping with all the suggestions and the testing over the past two months. All this work has resulted in a solid 1.84 release. I have enjoyed the coding side of this as there were several interesting problems to solve. I'll get this posted in the next day or two. We can start a separate discussion topic for what our next release, v1.85 might look like. Tnx and 73, Mike

**Comment by w7sst** [comment]
v1.84 is out the door. Thanks again to all the help from everyone. 73, Mike

**Comment by WR7Q** [comment]
Thanks Mike.

I like it a lot and you have done a terrific job on this.  I will be using
it from now on in all my CW Academy classes.

73,
Bob WR7Q
CWA Admin
